src/create-broadcast.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Creates a broadcast that can be listened to, i.e. simple event emitter
+ *
+ * @see https://github.com/ReactTraining/react-broadcast
+ */
+
+const createBroadcast = initialValue => {
+  let listeners = [];
+  let currentValue = initialValue;
+
+  return {
+    publish(value) {
+      currentValue = value;
+      listeners.forEach(listener => listener(currentValue));
+    },
+    subscribe(listener) {
+      listeners.push(listener);
+
+      // Publish to this subscriber once immediately.
+      listener(currentValue);
+
+      return () => {
+        listeners = listeners.filter(item => item !== listener);
+      };
+    },
+  };
+};
+
+module.exports = createBroadcast;
```

Comment:
In glamorous we have a slightly different implementation.
The main difference is that we do not publish to subscribers immediately after creating subscriptions.

This enables `ThemeProvider` to not create subscriptions in `componentWillMount` (which is the recommended behavior according to React docs), but instead leverages `componentDidMount` to setup subscription to the broadcaster, and `componentWillMount` to get the state from the broadcaster, and synchronously apply theming without triggering a re-render.

https://github.com/vesparny/brcast

cc @kentcdodds